Revit_Core_Engine/Create/FamilyInstance/FamilyInstance.cs

```diff
@@ -187,6 +189,11 @@ private static FamilyInstance FamilyInstance_OneLevelBased(Document document, Fa
             if (host != null)
                 familyInstance.HostIgnoredWarning();
 
+            double angle = XYZ.BasisX.AngleOnPlaneTo(orientation.ProjectedX(settings), XYZ.BasisZ);
+            XYZ axisEnd = new XYZ(origin.X, origin.Y, origin.Z + 30);
+            Autodesk.Revit.DB.Line axis = Autodesk.Revit.DB.Line.CreateBound(origin, axisEnd);
+            ElementTransformUtils.RotateElement(document, familyInstance.Id, axis, angle);
```

Comment:
This could be written a bit more concisely I believe
```suggestion
            ElementTransformUtils.RotateElement(document, familyInstance.Id, Line.CreateUnbound(origin, XYZ.BasisZ), angle);
```